app/assets/javascripts/map.js

```diff
@@ -0,0 +1,44 @@
+
+    var geocoder;
+    var map;
+    function initialize() {
+  geocoder = new google.maps.Geocoder();
+  }
+  function codeAddress() {
+  var address = document.getElementById('address').value;
+  geocoder.geocode( { 'address': address}, function(results, status) {
+    if (status == google.maps.GeocoderStatus.OK) {
+      map.setZoom(6);
+      map.setCenter(results[0].geometry.location);
+    } else {
+      alert('Geocode was not successful for the following reason: ' + status);
+    }
+  });
+}
+
+    var map = new google.maps.Map(document.getElementById('map'), {
+      zoom: 4,
+      center: new google.maps.LatLng(40.707437, -99.592151),
+      mapTypeId: google.maps.MapTypeId.ROADMAP,
+      disableDefaultUI: true
+    });
+
+    var infowindow = new google.maps.InfoWindow();
+
+    var marker, i;
+
+    for (i = 0; i < locations.length; i++) {  
+      marker = new google.maps.Marker({
+        position: new google.maps.LatLng(locations[i][1], locations[i][2]),
+        map: map
+      });
+
+      google.maps.event.addListener(marker, 'click', (function(marker, i) {
+        return function() {
+          infowindow.setContent(locations[i][0]);
+          infowindow.open(map, marker);
+        }
+      })(marker, i));
+    }
+
+    google.maps.event.addDomListener(window, 'load', initialize);

```

Comment:
looks like this needs a bit more work.

On Thu, Jun 18, 2015 at 12:34 AM, Hound <notifications@github.com> wrote:

> In app/assets/javascripts/map.js
> <https://github.com/OperationCode/operationcode/pull/58#discussion_r32704965>
> :
>
> > +
> > +    for (i = 0; i < locations.length; i++) {
> > +      marker = new google.maps.Marker({
> > +        position: new google.maps.LatLng(locations[i][1], locations[i][2]),
> > +        map: map
> > +      });
> > +
> > +      google.maps.event.addListener(marker, 'click', (function(marker, i) {
> > +        return function() {
> > +          infowindow.setContent(locations[i][0]);
> > +          infowindow.open(map, marker);
> > +        }
> > +      })(marker, i));
> > +    }
> > +
> > +    google.maps.event.addDomListener(window, 'load', initialize);
>
> 'google' is not defined.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/OperationCode/operationcode/pull/58/files#r32704965>.
>